### PR TITLE
Weapon ranking beta

### PR DESCRIPTION
**Pull request by 48klocs**
The folks over at destinytracker.net exposed an API for posting and fetching weapon ratings.  There's plenty more work to be done (item discussion's just a stub right now, UI is quaint, etc.), but grabbing average scores and posting reviews is in and useful.

There were some i18n additions.

	- Added to dim_en.json
		○ Settings
			§ AllowIdPostToDtr
			§ AllowIdPostToDtrHelp
		○ DtrReview
			§ Error
			§ Help
			§ Submit
			§ DiscussHint

### CODE REVIEW

File: src/scripts/services/dimStoreService.factory.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ function StoreService(
   dimManifestService,
   $translate,
   uuid2,
-  dimFeatureFlags
+  dimFeatureFlags,
+  dimDestinyTrackerService
```
Comment (48klocs) [inline src/scripts/services/dimStoreService.factory.js]: Without this dangling reference, the tracker service doesn't end up being instantiated.  Google couldn't provide me any guidance on writing what amounts to a background daemon in Angular.

File: src/scripts/itemDiscuss/dimItemDiscuss.directive.template.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div id="discuss-drawer" ng-if="vm.show">
+    <div class="review-list">Seek out other Guardians when the howling loneliness grows too deep. Pretend it's a chance meeting, though.</div>
```
Comment (delphiactual) [inline src/scripts/itemDiscuss/dimItemDiscuss.directive.template.html]: `<div class="review-list" translate="DtrReview.FlavorText"></div>`

File: src/scripts/services/dimDestinyTrackerService.factory.js [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+}
+
+class trackerErrorHandler {
+  constructor($q) {
+    this.$q = $q;
+  }
+
+  handleErrors(response) {
+    if (response.status !== 200) {
+      return this.$q.reject(new Error("Destiny tracker service call failed."));
```
Comment (delphiactual) [inline src/scripts/services/dimDestinyTrackerService.factory.js]: return this.$q.reject(new Error($translate.instant('DtrReview.ServiceError'));

File: src/scripts/services/dimDestinyTrackerService.factory.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    }
+
+    return response;
+  }
+
+  handleSubmitErrors(response) {
+    if (response.status !== 204) {
+      return this.$q.reject(new Error("Destiny tracker service submit failed."));
```
Comment (delphiactual) [inline src/scripts/services/dimDestinyTrackerService.factory.js]: return this.$q.reject(new Error($translate.instant('DtrReview.ServiceSubmitError'));

**Comment by bhollis** [comment]
Wow, super exciting! I'll try to check out the branch and peek at the UI soon.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # next
 
 * Fixed the perk selection in Loadout Builder. #1453
+* Integrated Trials-centric weapon reviews (and the ability to rate your own gear (and make comments about your gear)).  Done in conjunction with destinytracker.net.
```
Comment (sgtfrankieboy) [inline CHANGELOG.md]: It's destinytracker.com ;). Unless you used it way back in 2014 when it still was destinytracker.net.

Also you're saying gear, might be an idea to change that to weapon?

File: src/i18n/dim_en.json [hunk 6]
```diff
@@ -294,6 +303,8 @@
       "ShowAnimationsHelp": "Will show the animated new item overlay on new items. Turning this off can save CPU cycles.",
       "ShowElemental": "Show elemental damage icons on weapons",
       "ShowElementalHelp": "Show elemental damage icons on weapons.",
+      "AllowIdPostToDtr": "Allow potentially identifying item information to be posted to Destiny Tracker.",
+      "AllowIdPostToDtrHelp": "Allow DIM to post identifying item data to DTR.",
```
Comment (sgtfrankieboy) [inline src/i18n/dim_en.json]: Might also be an idea to mention which information can be send to Destiny Tracker.

- **InstanceId** _When getting Weapon Reviews, Submitting a Review. A unique ID specific to the player's version._
- **MembershipId** _When submitting a Review, Unique ID identifying the players account._
- **DisplayName** _When submitting a Review, Name of the players account (gamertag, PSN ID)._

**Comment by bhollis** [comment]
One thing we'll definitely want here is a feature flag we can use to turn this feature on and off. You can look at the other feature flags to see how that works :-)

**Comment by 48klocs** [comment]
This has been superseded by something fancier.

**Pull request closed without merging**